src/Workspaces/CSharp/Portable/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -125,9 +125,11 @@ private static bool BaseTypeDeclarationContainsPosition(BaseTypeDeclarationSynta
         private static readonly Func<SyntaxKind, bool> s_isDotOrArrowOrColonColon =
             k => k == SyntaxKind.DotToken || k == SyntaxKind.MinusGreaterThanToken || k == SyntaxKind.ColonColonToken;
 
-        public static bool IsRightOfDotOrArrowOrColonColon(this SyntaxTree syntaxTree, int position, CancellationToken cancellationToken)
+        public static bool IsRightOfDotOrArrowOrColonColon(this SyntaxTree syntaxTree, int position, SyntaxToken targetToken, CancellationToken cancellationToken)
```

Comment:
another option would be to simply not pass in the target token.  juts pass in the position.  then just handle the `..` case there if the `IsRightOf` check returns false.

Reply:
Thank you, @cyrusnajmabadi! I see this point. i considered it as well but found that we make those calculations in caller methods as well. So, let us not duplicate the in the callee.

---
In reply to: [254097365](https://github.com/dotnet/roslyn/pull/33148#discussion_r254097365) [](ancestors = 254097365)